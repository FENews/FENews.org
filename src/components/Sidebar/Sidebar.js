import React from 'react';
import { graphql, StaticQuery } from 'gatsby';
import Author from './Author';
import Contacts from './Contacts';
import Copyright from './Copyright';
import Menu from './Menu';
import styles from './Sidebar.module.scss';

export const PureSidebar = ({ data, isIndex }) => {
  const {
    author,
    copyright,
    menu
  } = data.site.siteMetadata;

  return (
    <div className={styles['sidebar']}>
      <div className={styles['sidebar__inner']}>
        <Author author={author} isIndex={isIndex} />
        <Menu menu={menu} />
        <Contacts contacts={author.contacts} />
        <Copyright copyright={copyright} />
      </div>
    </div>
  );
};

export const Sidebar = (props) => (
  <StaticQuery
    query={graphql`
      query SidebarQuery {
        site {
          siteMetadata {
            title
            subtitle
            copyright
            menu {
              label
              path
            }
            author {
              name
              photo
              bio
              contacts {       
                twitter
                telegram
                github
                email
                rss
                vkontakte
              }
            }
          }
        }
      }
    `}
    render={(data) => <PureSidebar {...props} data={data}/>}
  />
);

export default Sidebar;
