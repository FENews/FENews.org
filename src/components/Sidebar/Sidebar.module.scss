@import '../../assets/scss/variables';
@import '../../assets/scss/mixins';

.sidebar {
    width: 100%;
    &__inner {
        position: relative;
        padding: 25px 20px 0;
    }
}

@include breakpoint-sm {
    .sidebar {
        lost-column: 5/12;
        &__inner {
            padding: 30px 20px 0;
            &:after {
                background: $color-gray-border;
                background: linear-gradient(to bottom, $color-gray-border 0%, $color-gray-border 48%, $color-white 100%);
                position: absolute;
                content: '';
                width: 1px;
                height: 540px;
                top: 30px;
                right: -10px;
                bottom: 0;
            }
        }
    }
}

@include breakpoint-md {
    .sidebar {
        lost-column: 1/3;
        &__inner {
            padding: 40px;
        }
    }
}
