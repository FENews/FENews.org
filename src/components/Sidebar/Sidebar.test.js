import React from 'react';
import renderer from 'react-test-renderer';
import { PureSidebar as Sidebar } from './Sidebar';

describe('Sidebar', () => {
  it('renders correctly', () => {
    const props = {
      data: {
        site: {
          siteMetadata: {
            author: {
              name: 'name',
              photo: '/photo.jpg',
              bio: 'bio',
              contacts: {
                email: '#',
                twitter: '#',
                vkontakte: '#',
                github: '#',
                rss: '#',
                telegram: '#'
              }
            },
            copyright: 'copyright',
            menu: [
              {
                label: 'Item 0',
                path: '/#0/'
              },
              {
                label: 'Item 1',
                path: '/#1/'
              }
            ]
          }
        }
      }
    };

    const tree = renderer.create(<Sidebar {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
