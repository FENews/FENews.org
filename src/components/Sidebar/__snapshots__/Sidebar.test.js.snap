// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar renders correctly 1`] = `
<div
  className="sidebar"
>
  <div
    className="sidebar__inner"
  >
    <div
      className="author"
    >
      <Link
        to="/"
      >
        <img
          alt="name"
          className="author__photo"
          height="75"
          src="/photo.jpg"
          width="75"
        />
      </Link>
      <h2
        className="author__title"
      >
        <Link
          className="author__title-link"
          to="/"
        >
          name
        </Link>
      </h2>
      <p
        className="author__subtitle"
      >
        bio
      </p>
    </div>
    <nav
      className="menu"
    >
      <ul
        className="menu__list"
      >
        <li
          className="menu__list-item"
        >
          <Link
            activeClassName="menu__list-item-link--active"
            className="menu__list-item-link"
            to="/#0/"
          >
            Item 0
          </Link>
        </li>
        <li
          className="menu__list-item"
        >
          <Link
            activeClassName="menu__list-item-link--active"
            className="menu__list-item-link"
            to="/#1/"
          >
            Item 1
          </Link>
        </li>
      </ul>
    </nav>
    <div
      className="contacts"
    >
      <ul
        className="contacts__list"
      >
        <li
          className="contacts__list-item"
        >
          <a
            className="contacts__list-item-link"
            href="mailto:#"
            rel="noopener noreferrer"
            target="_blank"
          >
            <svg
              className="icon"
              viewBox="0 0 28 28"
            >
              <path
                d="M26 23.5v-12c-0.328 0.375-0.688 0.719-1.078 1.031-2.234 1.719-4.484 3.469-6.656 5.281-1.172 0.984-2.625 2.188-4.25 2.188h-0.031c-1.625 0-3.078-1.203-4.25-2.188-2.172-1.813-4.422-3.563-6.656-5.281-0.391-0.313-0.75-0.656-1.078-1.031v12c0 0.266 0.234 0.5 0.5 0.5h23c0.266 0 0.5-0.234 0.5-0.5zM26 7.078c0-0.391 0.094-1.078-0.5-1.078h-23c-0.266 0-0.5 0.234-0.5 0.5 0 1.781 0.891 3.328 2.297 4.438 2.094 1.641 4.188 3.297 6.266 4.953 0.828 0.672 2.328 2.109 3.422 2.109h0.031c1.094 0 2.594-1.437 3.422-2.109 2.078-1.656 4.172-3.313 6.266-4.953 1.016-0.797 2.297-2.531 2.297-3.859zM28 6.5v17c0 1.375-1.125 2.5-2.5 2.5h-23c-1.375 0-2.5-1.125-2.5-2.5v-17c0-1.375 1.125-2.5 2.5-2.5h23c1.375 0 2.5 1.125 2.5 2.5z"
              />
            </svg>
          </a>
        </li>
        <li
          className="contacts__list-item"
        >
          <a
            className="contacts__list-item-link"
            href="https://github.com/#"
            rel="noopener noreferrer"
            target="_blank"
          >
            <svg
              className="icon"
              viewBox="0 0 26 28"
            >
              <path
                d="M10 19c0 1.141-0.594 3-2 3s-2-1.859-2-3 0.594-3 2-3 2 1.859 2 3zM20 19c0 1.141-0.594 3-2 3s-2-1.859-2-3 0.594-3 2-3 2 1.859 2 3zM22.5 19c0-2.391-1.453-4.5-4-4.5-1.031 0-2.016 0.187-3.047 0.328-0.812 0.125-1.625 0.172-2.453 0.172s-1.641-0.047-2.453-0.172c-1.016-0.141-2.016-0.328-3.047-0.328-2.547 0-4 2.109-4 4.5 0 4.781 4.375 5.516 8.188 5.516h2.625c3.813 0 8.188-0.734 8.188-5.516zM26 16.25c0 1.734-0.172 3.578-0.953 5.172-2.063 4.172-7.734 4.578-11.797 4.578-4.125 0-10.141-0.359-12.281-4.578-0.797-1.578-0.969-3.437-0.969-5.172 0-2.281 0.625-4.438 2.125-6.188-0.281-0.859-0.422-1.766-0.422-2.656 0-1.172 0.266-2.344 0.797-3.406 2.469 0 4.047 1.078 5.922 2.547 1.578-0.375 3.203-0.547 4.828-0.547 1.469 0 2.953 0.156 4.375 0.5 1.859-1.453 3.437-2.5 5.875-2.5 0.531 1.062 0.797 2.234 0.797 3.406 0 0.891-0.141 1.781-0.422 2.625 1.5 1.766 2.125 3.938 2.125 6.219z"
              />
            </svg>
          </a>
        </li>
        <li
          className="contacts__list-item"
        >
          <a
            className="contacts__list-item-link"
            href="#"
            rel="noopener noreferrer"
            target="_blank"
          >
            <svg
              className="icon"
              viewBox="0 0 22 28"
            >
              <path
                d="M6 21c0 1.656-1.344 3-3 3s-3-1.344-3-3 1.344-3 3-3 3 1.344 3 3zM14 22.922c0.016 0.281-0.078 0.547-0.266 0.75-0.187 0.219-0.453 0.328-0.734 0.328h-2.109c-0.516 0-0.938-0.391-0.984-0.906-0.453-4.766-4.234-8.547-9-9-0.516-0.047-0.906-0.469-0.906-0.984v-2.109c0-0.281 0.109-0.547 0.328-0.734 0.172-0.172 0.422-0.266 0.672-0.266h0.078c3.328 0.266 6.469 1.719 8.828 4.094 2.375 2.359 3.828 5.5 4.094 8.828zM22 22.953c0.016 0.266-0.078 0.531-0.281 0.734-0.187 0.203-0.438 0.313-0.719 0.313h-2.234c-0.531 0-0.969-0.406-1-0.938-0.516-9.078-7.75-16.312-16.828-16.844-0.531-0.031-0.938-0.469-0.938-0.984v-2.234c0-0.281 0.109-0.531 0.313-0.719 0.187-0.187 0.438-0.281 0.688-0.281h0.047c5.469 0.281 10.609 2.578 14.484 6.469 3.891 3.875 6.188 9.016 6.469 14.484z"
              />
            </svg>
          </a>
        </li>
      </ul>
    </div>
    <div
      className="copyright"
    >
      copyright
    </div>
  </div>
</div>
`;
